src/query/src/part_sort.rs

```diff
@@ -664,7 +797,14 @@ mod test {
                 })
                 .collect_vec();
 
-            run_test(0, input_ranged_data, schema.clone(), opt, expected_output).await;
+            run_test(
```

Comment:
Consider adding some cases that concat batches from multiple ranges into one batch.

Reply:
00eb3a3